032_discord_only.py

```diff
@@ -323,77 +342,88 @@ async def maintainTwitchNotifs():
-
-                    ### TODO: get twitch user color and set in embed
-                    title = re.sub(r"[\r\n\t]", "", stream["Title"])
-                    game = stream["GameName"]
-                    embed = discord.Embed(title=f"Streaming {game}", description=f'\"{title}\"', color=0x009e00)
-                    embed.set_author(name=f"{stream['Username']}", url=f"https://haloruns.com/profiles/{stream['Username']}")
-                    ### TODO: Get Game Name from site when we get functionality to detect game.
-                    embed.add_field(name="\u200b", value=f"[Watch Here]({stream['StreamUrl'].lower()})", inline=True)
-                    stream_duration_raw = timedelta(seconds=math.ceil((datetime.now(timezone.utc) - parser.parse(stream['StartedAt'])).total_seconds() / 60) * 60)
-                    stream_duration = humanize.precisedelta(stream_duration_raw, minimum_unit="minutes", format="%.0f")
-                    embed.set_footer(text=f"{stream['Viewers']} Viewers | Streaming for {stream_duration}")
-                    ### Not sure if we want viewer count, but its here if we do.
-                    embed.set_thumbnail(url=stream['ProfileImageUrl'])
-                    embed.set_image(url=f"https://static-cdn.jtvnw.net/previews-ttv/live_user_{stream['TwitchUsername'].lower()}-640x360.jpg?ts={int(time.time())}")
-
-                    if existing_msg == None:
-                        print(f"{stream_url} not in: {streamMessages}")
-                        await notif_channel.send(embed=embed)
-                        currentStreams += 1
-                    else:
-                        # update the embed
-                        await existing_msg.edit(embed=embed)
-
-            if currentStreams > 0 and NOTICE_TEXT == "":
-                await topMessage.edit(content=SOME_STREAMS_TEXT)
-        else:
-            #logEvent("Throttled streams poll")
-            print(f"Throttled streams poll")
+        try:
+            await asyncio.sleep(10) # Timer to loop, better way but haven't gotten around to changing it
+            #todo: slow down traffic using lastUpdated - 1 minute intervals at least
+            #                                      datetime.datetime
+            diff = int( (datetime.now(timezone.utc) - updatedAt("streams") ).total_seconds())
+            if diff > STREAMS_THROTTLE:
+                logger.info("diff too high, calling streams")
+                logger.info("Looking for streams to post")
+                apiData = getJSON(STREAMS_ENDPOINT)
+
+                saveStreamList(apiData)
+
+                notif_channel = mb.get_channel(NOTIFS_CHANNEL_ID)
+                allMessages = await notif_channel.history(oldest_first=True).flatten()
+                if(len(allMessages) == 0):
+                    topMessage = await notif_channel.send(content=NO_STREAMS_TEXT)
+                else:
+                    topMessage = allMessages[0]

```

Comment:
This is the only change beyond indentations (which github annoyingly shows). This makes it so the bot does not assume a channel has a message which is not the case with an empty one.